src/Build/Construction/Solution/ProjectInSolution.cs

```diff
@@ -168,11 +168,21 @@ public string AbsolutePath
             {
                 if (_absolutePath == null)
                 {
+                    _absolutePath = Path.Combine(ParentSolution.SolutionFileDirectory, _relativePath);
+
+                    try
+                    {
 #if NETFRAMEWORK && !MONO
-                    _absolutePath = Path.GetFullPath(Path.Combine(ParentSolution.SolutionFileDirectory, _relativePath));
+                        _absolutePath = Path.GetFullPath(_absolutePath);
 #else
-                    _absolutePath = FileUtilities.NormalizePath(Path.Combine(ParentSolution.SolutionFileDirectory, _relativePath));
+                        _absolutePath = FileUtilities.NormalizePath(_absolutePath);
 #endif
+                    }
+                    catch (Exception)
+                    {
+                        // The call to GetFullPath can throw if the relative path is a URL or the paths are too long for the current file system
```

Comment:
Could we add a simpler path normalization here like (look for `\..\`) -> remove along with the prior path segment?

Reply:
I'd hate to get the logic wrong, I think the new code is pretty good to handle the situation more correctly.